# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/BytecodeCompiler.cpp
# Commit: 2ae31596c6a9
# Full Hash: 2ae31596c6a9a1699aadc0fd783cd94203d34bc2
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-12-05 09:58:04
# Regressor Bug: 1933681
# File Overlap Count: 1
# Description:
#   Bug 1933681 - Part 4: Remove duplicate maybeGetStencils call. r=nbp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231019
# ==============================================================================

diff -r 51797acbb256 -r 2ae31596c6a9 js/src/frontend/BytecodeCompiler.cpp
--- a/js/src/frontend/BytecodeCompiler.cpp	Wed Dec 04 23:44:34 2024 +0000
+++ b/js/src/frontend/BytecodeCompiler.cpp	Wed Dec 04 23:44:34 2024 +0000
@@ -1551,11 +1551,7 @@
     output.as<RefPtr<CompilationStencil>>() = std::move(stencil);
   } else {
     MOZ_ASSERT(maybeCx);
-    if (input.lazyOuterScript().sourceObject() &&
-        input.lazyOuterScript().sourceObject()->maybeGetStencils()) {
-      RefPtr<frontend::InitialStencilAndDelazifications> stencils =
-          input.lazyOuterScript().sourceObject()->maybeGetStencils();
-
+    if (stencils) {
       auto extensibleStencil =
           maybeCx->make_unique<frontend::ExtensibleCompilationStencil>(
               std::move(compilationState));
